CONTRIBUTING.md

```diff
@@ -74,6 +75,41 @@ With above, we can easily synchronize local branches with upstream branches.
 
 Right now we assume every contribution via pull request is for the `master` branch in Inclavare Containers. There are several other branches such as rc branches, release branches and backport branches. Before officially releasing a version, we may checkout a rc (release candidate) branch for more testings. When officially releasing a version, there may be a release branch before tagging which will be deleted after tagging. When backporting some fixes to existing released version, we will checkout backport branches.
 
+### C Code Style
+
+Inclavare Containers project uses `indent` to format C code style.
+
+1. Install `indent` using the following commands. Please make sure the version of `indent` must be 2.2.12 to get advanced features.
+
+```shell
+wget -c https://https://ftp.gnu.org/gnu/indent/indent-2.2.12.tar.gz
+tar xzf indent-2.2.12.tar.gz
+cd indent-2.2.12
+./configure
+make
+sudo make install
+```
+
+2. Format C code style in inclavare containers project using the following command:
+
+```shell
+indent -npro -linux -il-8 -ppi2 -nbbo -ci0 -cs -cp1 -gts foo.c
+```
+
+Because indent can't understand the C language grammar, there are problems with formatted code in some cases. For example, indent will completely destroy the indentation of the inline assembly because it does not recognize inline assembly statements.
+
+To avoid this problem, you need to [disable formatting](https://www.gnu.org/software/indent/manual/indent.html#SEC13) of C code for portions of a program by embedding special control comments in the program. For example, you can use `/* *INDENT-OFF* */` and `/* *INDENT-ON* */` to forbid `indent` to format the following inline assembly code.
+
+```shell
+/* *INDENT-OFF* */
+asm volatile(".byte 0x0f,0x01,0xd0" /* xgetbv */
+	     : "=a" (eax), "=d" (edx)
+	     : "c" (index));
+/* *INDENT-ON* */
+```
+
+Please follow [this guide](https://www.gnu.org/software/indent/manual/indent.html) for more information about `indent`.
```

Comment:
Please refer to [this manual](https://www.gnu.org/software/indent/manual/indent.html) for the details about the options of `indent`,